[ERP-1934] Update environs and other dependencies

nifadyev opened the pull request:
[[ERP-1934] [Yandex Tracker Action] Обновить пакет environs](https://tracker.yandex.ru/ERP-1934)

- environs<=10 uses marshmallow, but there was recent major update which removed some deprecated logic
As a result, environs was broken

nifadyev commented on requirements.txt:
```diff
@@ -1,19 +1,5 @@
+environs==14.1.1
+PyGithub==2.6.1
+requests==2.32.3
+# safety does not support version 4 for now
```
Согласен, тогда в рамках этого PR изменю

soltanoff replied:
Если `environs` сильно голову делает, то его тоже давайте выбрасывать в пользу старого доброго и надежного `os.getenv`

soltanoff commented on .github/workflows/pr.yaml:
```diff
@@ -10,9 +10,9 @@ jobs:
     timeout-minutes: 30
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
```
Увидел что ты там экшены руками апдейтил. Смотри какие опции есть для dependabot: 

```yaml
  - package-ecosystem: github-actions
    directory: "/"
    schedule:
      interval: daily
      time: "02:00"
    open-pull-requests-limit: 10
  - package-ecosystem: docker
    directory: "/"
    schedule:
      interval: daily
      time: "02:00"
    open-pull-requests-limit: 10
  - package-ecosystem: docker-compose
    directory: "/"
    schedule:
      interval: daily
      time: "02:00"
    open-pull-requests-limit: 10
```

nifadyev replied:
Ок, добавлю их, не хотелось раздувать PR большим числом изменений

nifadyev committed:
[ERP-1934] Update environs and other dependencies

- environs<=10 uses marshmallow but there was recent major update which removed some deprecated logic

As a result, environs was broken

- Replace safety with pip-audit for checking security vulnerabilities
